microsoft/PowerToys
# Fix ColorPicker detecting grid

## Summary of the Pull Request

This should fix the issue of sampling the grid in ColorPicker by first having the grid not draw inside the sampled pixel (https://github.com/microsoft/PowerToys/issues/13831#issuecomment-1019313016) and by trying to fix the timing of getting the colour (example here https://github.com/microsoft/PowerToys/issues/13831#issuecomment-1445470972).

The sampling period was 10ms, but that overlaps strangely with usual screen refresh periods (16.6ms, 13.3ms, etc) and can result in sampling the grid before the new one is drawn for multiple frames. This change is making the sampling period equal to the refresh period of primary display.

I've only tested it manually.

I haven't discussed this change myself with any of the core contributors, so I understand if it will be rejected. However, I thought the comments in the issue were enough to start working on it.


## PR Checklist

- [x] **Closes:** #13831


## Detailed Description of the Pull Request / Additional comments


## Validation Steps Performed

Ran ColorPicker at 100% and 125% scaling, at zoom levels where the grid is shown and moved the mouse quickly in random directions to try to get the app to sample the grid. Recorded the screen with an iPhone camera in slow-mo at 240fps for a minute to verify that it only picked the right colour.
